StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/SavedPaymentMethodFormFactory/SavedPaymentMethodFormFactory+Card.swift

```diff
@@ -13,40 +13,46 @@ import Foundation
-                guard let self = self else { return }
-                let selectedCardBrand = viewModel.selectedCardBrand ?? .unknown
-                let params = ["selected_card_brand": STPCardBrandUtilities.apiValue(from: selectedCardBrand), "cbc_event_source": "edit"]
-                STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: viewModel.hostedSurface.analyticEvent(for: .openCardBrandDropdown),
-                                                                     params: params)
-            } didTapClose: { [weak self] in
-                guard let self = self else { return }
-                let selectedCardBrand = viewModel.selectedCardBrand ?? .unknown
-                STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: viewModel.hostedSurface.analyticEvent(for: .closeCardBrandDropDown),
-                                                                     params: ["selected_card_brand": STPCardBrandUtilities.apiValue(from: selectedCardBrand)])
-            }
-
+                                                                               includePlaceholder: false)
             // pre-select current card brand
             if let currentCardBrand = viewModel.paymentMethod.card?.preferredDisplayBrand,
                let indexToSelect = cardBrandDropDown.items.firstIndex(where: { $0.rawData == STPCardBrandUtilities.apiValue(from: currentCardBrand) }) {
                 cardBrandDropDown.select(index: indexToSelect, shouldAutoAdvance: false)
-                viewModel.selectedCardBrand = currentCardBrand
             }
-            cardBrandDropDown.didUpdate = updateSelectedCardBrand
-            return cardBrandDropDown
-        }()
 
+            // Handler when user selects different card brand
+            let wrappedElement = PaymentMethodElementWrapper<DropdownFieldElement>(cardBrandDropDown){ field, params in
+                let cardBrands = self.viewModel.paymentMethod.card?.networks?.available.map({
+                    STPCard.brand(from: $0)
+                }).filter { self.viewModel.cardBrandFilter.isAccepted(cardBrand: $0) } ?? []
+                let cardBrand = cardBrands[field.selectedIndex]
+                let preferredNetworkAPIValue = STPCardBrandUtilities.apiValue(from: cardBrand)
+
+                params.paymentMethodParams.card?.networks = .init(preferred: preferredNetworkAPIValue)
+
+                if preferredNetworkAPIValue != self.lastCardBrandLogSelectedEventSent {
+                    STPAnalyticsClient.sharedClient.logPaymentSheetEvent(event: self.viewModel.hostedSurface.analyticEvent(for: .cardBrandSelected),
+                                                                         params: ["selected_card_brand": preferredNetworkAPIValue,
+                                                                                  "cbc_event_source": "edit", ])
+                    self.lastCardBrandLogSelectedEventSent = preferredNetworkAPIValue
```

Comment:
Is there a retain cycle here? The Factory class owns the card form, which owns this closure, which owns `self` (the factory).

Do we need to do this work in the closure at all?  Is this not work we can do in `didUpdate`?

Reply:
I booted up allocations and i'm not seeing any persistent allocations for SavedPaymentMethodFormFactory nor UpdatePaymentMethodViewModel.

Seems like the work should be done in here because it's updating the `params`.

In any case, seems like the factory should just create the instance and not own it. Will update.

Reply:
Made this class effectively a container for static methods, so this comment now seems out of date. Thanks for calling this out!